Misc. support for nrf9280pdk_cpuapp_iron

ayla-nordicsemi opened the pull request:
This pull request adds dts files required to support nrf9280pdk iron variant for the next Nordic secure firmware. 
Modifications are spread through Drivers, Firmware and Soc files.

ayla-nordicsemi committed:
boards: nordic: add 'iron' board variant to application

- Add support of next generation secure firmware to nrf92 application core.

Signed-off-by: Aymen LAOUINI <aymen.laouini@nordicsemi.no>

ayla-nordicsemi committed:
soc: nordic: add support of 'iron' variant to application

- use early init hook instead of nrf92_init.
- add config flag for iron variant.

Signed-off-by: Aymen LAOUINI <aymen.laouini@nordicsemi.no>

ayla-nordicsemi committed:
drivers: firmware: nrf_ironside: add nrf9280 to CPUCONF service

CPUCONF service now supports application on nrf92 application cores.

Signed-off-by: Aymen LAOUINI <aymen.laouini@nordicsemi.no>

ayla-nordicsemi committed:
drivers: firmware: nrf_ironside: Update doxygen header for API.

- Message is written to the boot report of core.

Signed-off-by: Aymen LAOUINI <aymen.laouini@nordicsemi.no>

nordicjm commented on boards/nordic/nrf9280pdk/Kconfig.defconfig:
```diff
@@ -19,3 +19,13 @@ config ASSERT
 
 endif # BOARD_NRF9280PDK_NRF9280_CPUPPR
+
+if BOARD_NRF9280PDK_NRF9280_CPUAPP_IRON
```
soc commit needs to come before board commit which uses said soc commit

nordicjm commented on boards/nordic/nrf9280pdk/nrf9280pdk_nrf9280-memory_map_iron.dtsi:
```diff
@@ -0,0 +1,112 @@
+
+/ {
+	reserved-memory {
+        cpuapp_cpusys_ipc_shm: memory@2f88f600 {
```
missing indent

nordicjm commented on boards/nordic/nrf9280pdk/nrf9280pdk_nrf9280-memory_map_iron.dtsi:
```diff
@@ -0,0 +1,112 @@
+			reg = <0x2f88fd00 0x200>;
+		};
+
+/* This cpurad section is to be removed soon as there is no need for it anymore.
+ * Removing it now causes to much issues to resolve, keeping it is simpler 
+ * and faster .. 
+ */
```
I don't follow - if you remove it, it simply isn't present, that shouldn't cause any issues, nothing checks that a node is 100% used with child nodes

nordicjm commented on boards/nordic/nrf9280pdk/Kconfig.defconfig:
```diff
@@ -19,3 +19,13 @@ config ASSERT
+config ROM_START_OFFSET
+	default 0x800 if BOOTLOADER_MCUBOOT
+
+config FLASH_LOAD_OFFSET
+	default 0x2c000 if !USE_DT_CODE_PARTITION
```
use dt Kconfig functions to get this value

anangl commented on boards/nordic/nrf9280pdk/Kconfig.nrf9280pdk:
```diff
@@ -2,7 +2,14 @@
 config BOARD_NRF9280PDK
-	select SOC_NRF9280_CPUAPP if BOARD_NRF9280PDK_NRF9280_CPUAPP
+	select SOC_NRF9280_CPUAPP if (BOARD_NRF9280PDK_NRF9280_CPUAPP || \
+					 BOARD_NRF9280PDK_NRF9280_CPUAPP_IRON)
```
alignment could be improved here

anangl commented on boards/nordic/nrf9280pdk/Kconfig.nrf9280pdk:
```diff
@@ -2,7 +2,14 @@
-	select SOC_NRF9280_CPUPPR if BOARD_NRF9280PDK_NRF9280_CPUPPR || \
-				     BOARD_NRF9280PDK_NRF9280_CPUPPR_XIP
+	select SOC_NRF9280_CPUPPR if (BOARD_NRF9280PDK_NRF9280_CPUPPR || \
+				     BOARD_NRF9280PDK_NRF9280_CPUPPR_XIP)
```
and here

anangl commented on boards/nordic/nrf9280pdk/nrf9280pdk_nrf9280_cpuapp_iron.yaml:
```diff
@@ -0,0 +1,23 @@
+# Copyright (c) 2024 Nordic Semiconductor ASA
```
2025

anangl commented on boards/nordic/nrf9280pdk/nrf9280pdk_nrf9280_cpuapp_iron_defconfig:
```diff
@@ -0,0 +1,29 @@
+# Copyright (c) 2024 Nordic Semiconductor ASA
```
2025

ayla-nordicsemi committed:
tests: drivers: watchdog: Enable watchdog test for nrf9280 iron

Added overlay for nrf9280 cpuapp iron board.

Signed-off-by: Aymen LAOUINI <aymen.laouini@nordicsemi.no>

ayla-nordicsemi committed:
tests: drivers: adc: Enable adc test for nrf9280 iron

Added overlay for nrf9280 cpuapp iron board.
Modified overlay for nrf9280 cpuapp board to use common dtsi.

Signed-off-by: Aymen LAOUINI <aymen.laouini@nordicsemi.no>

ayla-nordicsemi committed:
samples: watchdog: Add nrf9280 iron to sample

Added overlay for nrf9280 cpuapp iron board.

Signed-off-by: Aymen LAOUINI <aymen.laouini@nordicsemi.no>

nordicjm commented on soc/nordic/nrf92/Kconfig.soc:
```diff
@@ -62,5 +62,20 @@ config SOC_NRF9280_CPUPPR
+	help
+	  Generic Nordic IRONside SE for Series nRF92X.
+
+config SOC_SERIES_NRF92X_CPUAPP
+	bool
+	help
+	  Generic Nordic application core for Series nRF92X.
```
neither is this

nordicjm commented on boards/nordic/nrf9280pdk/Kconfig.nrf9280pdk:
```diff
@@ -2,7 +2,14 @@
+	select SOC_NRF9280_CPUPPR if (BOARD_NRF9280PDK_NRF9280_CPUPPR || \
+				       BOARD_NRF9280PDK_NRF9280_CPUPPR_XIP)
+	select SOC_NRF9280_IRON if BOARD_NRF9280PDK_NRF9280_CPUAPP_IRON
+	select SOC_NRF9280_IRON if BOARD_NRF9280PDK_NRF9280_CPUAPP_IRON
```
```suggestion
```

ayla-nordicsemi committed:
tests: kernel: gen_isr_table: Add ISR offset definitions for nrf9280

- Correct test by adding ISR offset definitions for nrf9280 ppr.

Signed-off-by: Aymen LAOUINI <aymen.laouini@nordicsemi.no>

ayla-nordicsemi committed:
tests: kernel: interrupt: Add ISR offset definitions for nrf9280

- Correct test by adding ISR offset definitions for nrf9280 ppr.

Signed-off-by: Aymen LAOUINI <aymen.laouini@nordicsemi.no>

carlescufi commented:
@nordicjm @mparpala @anangl @nika-nordic @kl-cruz please take a look
